DataFormats/PatCandidates/interface/Photon.h

```diff
@@ -169,6 +170,11 @@ namespace pat {
 	    //<< key << " was not stored for this particle.";
           }
       }
+
+      float puppiChargedHadronIso() const {return puppiChargedHadronIso_; };
+      float puppiNeutralHadronIso() const {return puppiNeutralHadronIso_; };
+      float puppiPhotonIso() const {return puppiPhotonIso_; };
+
```

Comment:
why is there no NoLeptons version?
Was it forgotten or is there a subtle reason why it should not be here?

IIUC, the main feature of "NoLeptons" from perspective of ChargedHadronIso is that the leptons are not used in the input to the puppi algo (do not trigger de-weighting of a given particle as being consistent with pileup ... somewhat assuming that the leptons are not pileup).
So, from this, if it makes sense to do "NoLeptons" for electrons, it should be just about as meaningful for photons.